cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java

```diff
@@ -130,4 +135,33 @@ private String getEndpointFromHost(String host) {
     segments.remove(0);
     return Joiner.on(".").join(segments);
   }
+
+  /**
+   *
+   * @param parameters JsonObject containing ms.source.s3.parameters
+   * @param host hostname with bucket name in the beginning
+   * @return the bucket name
+   */
+  @VisibleForTesting
+  protected String getBucketName(JsonObject parameters, String host) {
+    if (parameters.has(KEY_BUCKET_NAME)) {
+      return parameters.get(KEY_BUCKET_NAME).getAsString();
+    }
+    return host.split("\\.")[0];
+  }
+
+  /**
+   *
+   * @param parameters JsonObject containing ms.source.s3.parameters
+   * @param host hostname with bucket name in the beginning
+   * @return the endpoint name without the bucket name
+   */
+  @VisibleForTesting
+  protected String getEndpoint(JsonObject parameters, String host) {
+    if (parameters.has(KEY_BUCKET_NAME)) {
+      String bucketName = parameters.get(KEY_BUCKET_NAME).getAsString().toLowerCase();
+      host = host.toLowerCase(Locale.ROOT).replace(bucketName, "");
```

Comment:
minor concern butt I don't see the tolowercase used in the original getEndpointFromHost method. Is there an existing use case that has issues with cases?

Reply:
We are replacing bucket name from the host, if some client enters bucket name in different case as in the source uri. It won't be replaced properly.